cmd/azqr/main.go

```diff
@@ -119,6 +132,68 @@ func main() {
 	}
 }
 
+// ReviewContext A running resource group analysis support context
+type ReviewContext struct {
+	// Review context, will be passed to every created goroutines
+	Ctx context.Context
+	// Communication interface for each review results
+	ResCh chan []analyzers.AzureServiceResult
+	// Communication interface for errors
+	ErrCh chan error
+}
+
+// Run a review on a peculiar resource group "r" with the appropriates analysers using "concurrency" goroutines
+func reviewRunner(rc *ReviewContext, r string, svcAnalysers *[]analyzers.AzureServiceAnalyzer, concurrency int) {
+	if concurrency <= 0 {
+		concurrency = len(*svcAnalysers)
+	}
+	sem := semaphore.NewWeighted(int64(concurrency))
+	for i := range *svcAnalysers {
+		if err := sem.Acquire(rc.Ctx, 1); err != nil {
+			rc.ErrCh <- err
+			return
+		}
+		// When starting a goroutine from a loop, we cannot directly use
+		// the iteration variable, as only the last element of the loop will
+		// be processed
+		analyserPtr := &(*svcAnalysers)[i]
+		go func(a *analyzers.AzureServiceAnalyzer, r string) {
+			defer sem.Release(1)
+			// In case the analysis was cancelled, we don't need to execute the review
+			if context.Canceled == rc.Ctx.Err() {
+				return
+			}
+			res, err := (*a).Review(r)
+			if err != nil {
+				rc.ErrCh <- err
+			}
+			rc.ResCh <- res
+		}(analyserPtr, r)
+	}
+}
+
+// Wait for at least "nb" goroutines to hands their result and return them
+func waitForReviews(rc *ReviewContext, nb int) (*[]analyzers.AzureServiceResult, error) {
+	received := 0
+	reviews := make([]analyzers.AzureServiceResult, nb)
```

Comment:
if `nb` is used here it will render empty results.

```suggestion
	reviews := make([]analyzers.AzureServiceResult, 0)
```